subprojects/packagefiles/openal-soft/meson.build

```diff
@@ -6,9 +6,20 @@ project(
   default_options: ['cpp_std=c++14', 'c_std=c11'],
 )
 
+
+openal_all_dependencies = [ ]
+
 cpp = meson.get_compiler('cpp')
+
+# Some systems may need libatomic for atomic functions to work
+if cpp.compiles('''#include <atomic>
+std::atomic<int> foo{0};
+int main() { return foo.fetch_add(2); }''')
+  atomic_dep = cpp.find_library('atomic', required: true)
+  openal_all_dependencies += atomic_dep
+endif
```

Comment:
You can use `dependency('', required: false)` as a fallback instead:
```suggestion
# Some systems may need libatomic for atomic functions to work
atomic_dep = dependency('', required: false)
if cpp.compiles('''#include <atomic>
std::atomic<int> foo{0};
int main() { return foo.fetch_add(2); }''')
  atomic_dep = cpp.find_library('atomic', required: true)
endif
```
And remove all your `openal_all_dependencies` changes and unconditionally add `atomic_dep` to `openal_lib`'s dependencies.